src/utils/api.ts

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-console */
+
+import axios, { AxiosResponse } from 'axios'
+
+import { paths } from './api-schema'
+
+const instance = axios.create({
+  baseURL: process.env.REACT_APP_API_URL,
+  params: {
+    api_key: process.env.REACT_APP_API_KEY,
+  },
+})
+
+const handleResponse = (request: Promise<AxiosResponse<any>>) =>
+  request.then(
+    (res) => {
+      // v0.2 API handles client-side parameter errors with
+      // a 200 response and an error string in the body
+      if (res.data.error) {
+        console.error(res.data.error)
+        return null
+      }
+      return res.data
+    },
+    (err) => {
+      console.error(err)
+      return null
+    },
+  )
+const fileToFormData = (photo_data: string | Blob | undefined) => {
+  if (photo_data !== undefined) {
+    var formData = new FormData();
+    return formData.append("photo_data", photo_data)
+  } 
+  return null;
+  
+};
+
+// Follow this example!
+export const getClusters = (
+  params: paths['/clusters.json']['get']['parameters']['query'],
+) =>
+  handleResponse(
+    instance.get('/clusters.json', {
+      params,
+    }),
+  )
+
+export const getLocations = (
+  params: paths['/locations.json']['get']['parameters']['query'],
+) =>
+  handleResponse(
+    instance.get('/locations.json', {
+      params,
+    }),
+  )  
+
+
+export const postLocations = (
+  params: paths['/locations.json']['post']['parameters']['query'],
+  photo_data?: File,
+) => {
```

Comment:
You can remove the "body"/braces from this function by putting `fileToFormData(photo_data)` directly as the parameter.